core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -105,6 +107,14 @@ private[spark] class CoarseGrainedExecutorBackend(
       executor.stop()
       context.stop(self)
       context.system.shutdown()
+
+    // Add new credentials received from the driver to the current user.
+    case UpdateCredentials(newCredentials) =>
```

Comment:
Yes, I think that makes sense. We could just write it out to a new file on HDFS and send the path to the file via Akka.